src/main/studio/qeditor/RSTokenMaker.java

```diff
@@ -63,7 +63,7 @@ public Token getTokenList(Segment text, int initialTokenType, int startOffset) {
 
     @Override
     public boolean getCurlyBracesDenoteCodeBlocks(int languageIndex) {
-        return true;
+        return false;   //this needs to be false, otherwise the closing brace of functions will jump to the beginning of the line, which results in a syntax error
```

Comment:
I thought may be we can somehow override the algorithm of adding curling braces to add at least one space if comes on the beginning of the line?